### DISCUSSION TITLE
MONAI on AMD Instinct GPU

### DISCUSSION THREAD

**Discussion by pjaiuser** [Project-MONAI/MONAI#6746] [General]
Hello,
Can MONAI Label leverage AMD GPU's or work with NVIDIA and CUDA libraries only?

**Comment by KumoLiu** [comment]
Hi @pjaiuser, FYI https://github.com/Project-MONAI/MONAI/pull/6161.
We already add supports for AMD GPU but may still need more test.
Thanks!

**Comment by pjaiuser** [comment]
Thanks Yun for your prompt responses. But I don't see any documentation
around it. Is it because it's still under test until and won't be available
until it reaches its confidence level?

**Reply by wyli** [reply]
cc @yaomingamd

**Comment by yaomingamd** [comment]
Hi @pjaiuser
Yes, MONAI core, label, and deploy  support  AMD GPUs. We will add documentation soon.  To install MONAI on AMD ROCm platform, you need to install pytorch which supports AMD GPU at the first, then install MONAI. For Example, if your host ROCm version is ROCm 5.4.2, you can use this command to install pytorch : "pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/rocm5.4.2. ", then you can use normal command to install MONAI core from source:   FORCE_CUDA=1 BUILD_MONAI=1 pip install --no-build-isolation git+https://github.com/Project-MONAI/MONAI#egg=monai.  Here FORCE_CUDA, just means force GPU. Please  let me know if you have more questions and issues .

**Reply by pjaiuser** [reply]
Thanks @yaomingamd . My main objective is to deploy MONAI LABEL/CORE along with sample application and dataset and run the model on my server equipped with AMD EPYC processor and AMD GPU MI210. Hope this objective can be met.

**Reply by pjaiuser** [reply]
Also, do you know the ROCM, pytorch and OS version compatibility for MONAI or it is agnostic of this?

**Reply by pjaiuser** [reply]
My other question is tied to the objective. i want to showcase a comparison between CPU and GPU using MONAI, can it be run on AMD CPU based system or GPU is mandatory?

**Reply by yaomingamd** [reply]
@pjaiuser 
answer to your questions: 
Question 1:  Yes, you can.  AMD CI is running AMD Platform with AMD CPU and GPU (MI100 and MI200).  We also have MONAI docker container on AMD Accelerator Cloud (AAC:  aac.amd.com)  for customer to run MONAI on AMD hardware.  Sample applications from MONAI tutorials with medical datasets from medicaldecathlon.com  run on AMD GPUs successfully.  

Question 2:  For current dev branch of MONAI, we suggest ROCm pytorch >=2.0:
                    pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/rocm5.4.2 for ROCm 5.4
                    pip3 install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/rocm5.5 for ROCm 5.5
              and pip3 install torch==2.0.1 torchvision==0.15.2 -f https://repo.radeon.com/rocm/manylinux/rocm-rel-5.6/ for ROCm 5.6 

Question 3:   if your MONAI ML workflow runs on other CPU only systems, then it will run on AMD CPU only system too.
